src/util/fields.py

```diff
@@ -833,6 +833,7 @@ def _update_field_usage_multi(cr, models, old, new, domain_adapter=None, skip_in
 
     p = {
         "old": r"\y%s\y" % (re.escape(old),),
+        "old_pattern": r"""[\.'"]{0}\y""".format(re.escape(old)),
```

Comment:
No need to escape `.` in `[]`.
```suggestion
        "old_pattern": r"""[.'"]{0}\y""".format(re.escape(old)),
```

Reply:
```
[local] chs@t=# select 'blah.foo_id[0]' ~ '[.''"]foo_id\y';
┌──────────┐
│ ?column? │
├──────────┤
│ t        │
└──────────┘
(1 row)

Time: 4.612 ms
```